source/connect/server-selection.txt

```diff
@@ -0,0 +1,124 @@
+.. _pymongo-server-selection:
+
+==========================
+Customize Server Selection
+==========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, read preference, write
+
+Overview
+--------
+
+All MongoDB drivers follow a defined algorithm when selecting a server to read or write
+from. By using the ``server_selector`` property of ``MongoClient``, you can customize this
+algorithm to choose the server that works best for your application.
+
+.. important::
+
+   Customizing the server-selection algorithm can have unintended consequences,
+   such as degraded read or write performance.
+
+Customized Selection Algorithm
+------------------------------
+
+When {+driver-short+} executes a read operation, it performs the following steps,
+in order, to select a MongoDB deployment:
+
+1. From the list of known servers, {+driver-short+} selects all servers
+   that match the active read preference.
+
+#. If at least one readable server exists, {+driver-short+} calls the user-defined
+   server-selector function and passes in the list from the previous step.
+
+#. {+driver-short+} applies the ``localThresholdMS`` connection setting to the list of
+   servers returned from the function.
+
+#. {+driver-short+} selects a server at random from the servers still on the list and
+   executes the operation against this server.
+
+When {+driver-short+} executes a write operation, it begins by selecting all writeable
+servers, not just those that match the active read preference. The remaining steps are
+identical.
```

Comment:
I think this would be good in a .. note::

Reply:
I thought about it, but I went to this working group about callouts the other day. The consensus seemed to be that all teams use callouts too much, and "note" is better for like 'incidental' info rather than something core to the topic. I'm going to leave as is if that's cool